doc/ir-semantics.text

```diff
@@ -187,9 +183,7 @@ Update (partial) function, returns an H:
          Df  = Function f^ (formal*)
          I' := P(Df, lᵥ^)
          l' := l^
-         M  := (materialize P H E osr-map)*
-         H' := H ∪ range(M)
-         E' := (osr-env P H E M osr-map)*
+         (H', E') := sum (create-env H E osr-map)*
```

Comment:
I think that might help, but we also have partial environments (see line 157, `E+E'`).

`H+H'` looks weird to me. Updating the heap (and environment) directly would fix this, but then `create-env` gets uglier. Probably not worth it, especially when we have to call `create-env` for each `osr-map` or argument.